python/foglamp/tasks/north/sending_process.py

```diff
@@ -374,884 +284,654 @@ def __init__(self, loop=None):
+                .WHERE(['description', '=', self._name]) \
+                .payload()
+            streams = await self._storage_async.query_tbl_with_payload("streams", payload)
+            rows = streams['rows']
+            return rows[0]['id']
 
-        SendingProcess._logger.debug("{f} - update position - last_object/sent |{last}/{sent}| ".format(
-            f="_update_position_reached",
-            last=update_last_object_id,
-            sent=tot_num_sent))
+        stream_id = None
+        try:
+            payload = payload_builder.PayloadBuilder() \
+                .SELECT("id", "description", "active") \
+                .WHERE(['description', '=', self._name]) \
+                .payload()
+            streams = await self._storage_async.query_tbl_with_payload("streams", payload)
+            rows = streams['rows']
+            if len(rows) == 0:
+                stream_id = await add_stream(destination_id, self._name)
+                stream_id_valid = True
+            elif len(rows) > 1:
+                raise ValueError(_MESSAGES_LIST["e000013"].format(stream_id))
+            else:
+                stream_id = rows[0]['id']
+                if rows[0]['active'] == 't':
+                    stream_id_valid = True
+                else:
+                    SendingProcess._logger.info(_MESSAGES_LIST["i000004"].format(stream_id))
+                    stream_id_valid = False
+        except Exception as e:
+            SendingProcess._logger.error(_MESSAGES_LIST["e000013"].format(str(e)))
+            raise e
+        return stream_id, stream_id_valid
 
-        await self._last_object_id_update(update_last_object_id, stream_id)
+    async def _get_destination_id(self, destination_type, description):
+        async def add_destination(destination_type, description):
+            payload = payload_builder.PayloadBuilder() \
+                .INSERT(type=destination_type,
+                        description=description) \
+                .payload()
+            await self._storage_async.insert_into_tbl("destinations", payload)
+            payload = payload_builder.PayloadBuilder() \
+                .SELECT("id", "type", "description") \
+                .WHERE(['type', '=', destination_type]) \
+                .LIMIT(1) \
+                .payload()
+            destinations = await self._storage_async.query_tbl_with_payload("destinations", payload)
+            rows = destinations['rows']
+            return rows[0]['id']
 
-        await self._update_statistics(tot_num_sent, stream_id)
+        try:
+            payload = payload_builder.PayloadBuilder() \
+                .SELECT("id", "type", "description") \
+                .WHERE(['type', '=', destination_type]) \
+                .LIMIT(1) \
+                .payload()
+            destinations = await self._storage_async.query_tbl_with_payload("destinations", payload)
+            rows = destinations['rows']
```

Comment:
```
payload = payload_builder.PayloadBuilder() \
+                .SELECT("id", "type", "description") \
+                .WHERE(['type', '=', destination_type]) \
+                .LIMIT(1) \
+                .payload()
+            destinations = await self._storage_async.query_tbl_with_payload("destinations", payload)
+            rows = destinations['rows']
```
use above code snippet in def ? to avoid duplicate code